### DISCUSSION TITLE
如何使用cmake macro

### DISCUSSION THREAD

**Discussion by xiangyi-wang** [xmake-io/xmake#2405] [Q&A] [answered]
有一个项目是由多个子项目组成的([MPAS-JEDI](https://github.com/JCSDA/mpas-bundle)），基于cmake定义了一套宏命令作为整个项目的构建系统([ecbuild](https://github.com/ecmwf/ecbuild/blob/develop/bin/ecbuild))，我想对项目添加新模块，以及最后总的构建都改用xmake构建，而已有的子模块仍然使用这套宏构建，不作修改，该如何调用cmake的这些宏命令并把依赖的库传递给cmake构建系统

**Comment by waruqi** [comment, marked as answer]
cmake 我不了解，你可以百度下 cmake 里面的用法。。如果只是在 cmake 里面用 xmake 的依赖包，可以看下这个 https://github.com/xmake-io/xrepo-cmake

**Reply by xiangyi-wang** [reply]
好的 谢谢
